src/components/StandardPage/StandardPage.tsx

```diff
@@ -0,0 +1,144 @@
+  namespace: string;
+  RowMapper: React.FunctionComponent<RowProps<T>>;
+  supportedFilters?: {
+    [type: string]: (props: FieldFilterProps) => JSX.Element;
+  };
+  title: string;
+}
+
+export function StandardPage<T>({
+  namespace,
+  dataSource: [flattenData, loaded, error],
+  RowMapper,
+  title,
+  addButton,
+  fieldsMetadata,
+  supportedFilters = {
+    enum: EnumFilter,
+    freetext: FreetextFilter,
+  },
+}: StandardPageProps<T>) {
+  const { t } = useTranslation();
+  const [selectedFilters, setSelectedFilters] = useState({});
+  const clearAllFilters = () => setSelectedFilters({});
+  const [fields, setFields] = useFields(namespace, fieldsMetadata);
+
+  const filteredData = flattenData.filter(
+    createMetaMatcher(selectedFilters, fields),
+  );
+
+  const errorFetchingData = loaded && error;
+  const noResults = loaded && !error && flattenData.length == 0;
+  const noMatchingResults =
+    loaded && !error && filteredData.length === 0 && flattenData.length > 0;
+
+  return (
+    <>
+      <PageSection variant="light">
+        <Level>
+          <LevelItem>
+            <Title headingLevel="h1">{title}</Title>
+          </LevelItem>
+          {addButton && <LevelItem>{addButton}</LevelItem>}
+        </Level>
+      </PageSection>
+      <PageSection>
+        <Toolbar
+          clearAllFilters={clearAllFilters}
+          clearFiltersButtonText={t('Clear all filters')}
+        >
+          <ToolbarContent>
+            <ToolbarToggleGroup toggleIcon={<FilterIcon />} breakpoint="xl">
+              <PrimaryFilters
+                filterTypes={
+                  fields.filter(
+                    (field) => field.filter?.primary,
+                  ) as FilterType[]
+                }
+                onFilterUpdate={setSelectedFilters}
+                selectedFilters={selectedFilters}
+                supportedFilters={supportedFilters}
+              />
+              <AttributeValueFilter
+                filterTypes={
+                  fields.filter(
+                    ({ filter }) => filter && !filter.primary,
+                  ) as FilterType[]
+                }
+                onFilterUpdate={setSelectedFilters}
+                selectedFilters={selectedFilters}
+                supportedFilters={supportedFilters}
+              />
+              <ManageColumnsToolbar
+                columns={fields}
+                defaultColumns={fieldsMetadata}
+                setColumns={setFields}
+              />
+            </ToolbarToggleGroup>
+          </ToolbarContent>
+        </Toolbar>
+        <TableView<T>
+          entities={filteredData}
+          allColumns={fields}
+          visibleColumns={fields.filter(({ isVisible }) => isVisible)}
+          aria-label={title}
+          Row={RowMapper}
+        >
+          {[
+            !loaded && <Loading key="loading" />,
+            errorFetchingData && <ErrorState key="error" />,
+            noResults && <NoResultsFound key="no_result" />,
```

Comment:
usually we will have custom `NoResultsFound` component a user will want to put here, adding an optional prop named `customNoResultsFound` can give this functionality:

```suggestion
            noResults && customNoResultsFound && <NoResultsFound key="no_result" />,
```

Reply:
same for no matching